File: src/main/java/io/github/takenoha/towerdefense/persistence/CoreMutationResult.java
package io.github.takenoha.towerdefense.persistence;

import java.util.Objects;
import java.util.Optional;

/** Result of a UUID-protected core repair, move, or replacement. */
public record CoreMutationResult(
        ManagementOutcome outcome,
        Optional<CoreRecord> core) {
    public CoreMutationResult {
        Objects.requireNonNull(outcome, "outcome");
        Objects.requireNonNull(core, "core");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/persistence/CoreMutationResult.kt
package io.github.takenoha.towerdefense.persistence

import java.util.Optional
import kotlin.jvm.JvmRecord

/** Result of a UUID-protected core repair, move, or replacement. */
@JvmRecord
data class CoreMutationResult(
    val outcome: ManagementOutcome,
    val core: Optional<CoreRecord>,
)
